intranet/apps/dataimport/management/commands/delete_users.py

```diff
@@ -0,0 +1,88 @@
+import csv
+import sys
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = "Given a list of FCPS student IDs, delete them if they exist."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--filename", dest="filename", type=str, help="Filename to import data from. Must supply either this or 'student-id'.")
+        parser.add_argument(
+            "--column-header",
+            dest="header",
+            default="Student ID",
+            type=str,
+            nargs=1,
+            help="Header associated with the student ID column. Use with the 'filename' option",
+        )
+        parser.add_argument(
+            "--student-id", dest="student_ids", nargs="+", help="A space-separated list of student IDs to remove. Must supply this or 'filename'."
+        )
+        parser.add_argument("--run", action="store_true", dest="run", default=False, help="Actually run.")
+        parser.add_argument("--confirm", action="store_true", dest="confirm", default=False, help="Skip confirmation.")
+
+    def ask(self, q):
+        if input("{} [Yy]: ".format(q)).lower() != "y":
+            self.stdout.write(self.style.ERROR("Abort."))
+            sys.exit()
+
+    def read_student_ids(self, filename: str, column_header: str):
+        """Read student IDs from the given filename.
+
+        Args:
+            filename: Filename of the CSV to read from.
+            column_header: Column header of the Student ID column.
+
+        Returns:
+            A list of user IDs.
+
+        """
+        ids = []
+        with open(filename, "r") as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                ids.append(row[column_header])
+        return ids
+
+    def handle(self, *args, **options):
+        # Sanity check
+        if options["student_ids"] is None and options["filename"] is None:
+            raise ValueError("Must provide one of filename or student_id")
+
+        # Read in the list of student IDs, either from args or from the given filename
+        if options["student_ids"] is None:
+            ids = self.read_student_ids(options["filename"], options["header"])
+        else:
+            ids = options["student_ids"]
+
+        # Get a list of user model instances
+        users = []
+        for student_id in ids:
+            user = get_user_model().objects.filter(student_id=student_id).first()
```

Comment:
Why are you using `filter()` here and not `get()`? `student_id` is unique, so I think `get()` is more appropriate.

Reply:
My reasoning was that `get()` would raise a `DoesNotExist` if the given user(s) do not exist, which isn't the case with the data we're working with (some given users were already deleted).

From what I've seen elsewhere, the two ways to deal with this problem are a try-catch or the `filter()` that I used.

Reply:
You can also do .filter(student_id__in=ids) which will just return a QuerySet of all the students who have a student_id in ids

Reply:
You can also do .filter(student_id__in=ids) which will just return a QuerySet of all the students who have a student_id in ids

Reply:
Yea that’s probably the best way to minimize queries.